test/hermes/date-constructor.js

```diff
@@ -204,14 +212,60 @@ print(Date.parse('2016T12:30:47.1234-07:00'));
 // CHECK-NEXT: 1451676647123
 print(Date.parse('2016T12:30:47.760738998-07:00'));
 // CHECK-NEXT: 1451676647760
+print(Date.parse('2016T12:30:47.760738998-0700'));
+// CHECK-NEXT: 1451676647760
 print(Date.parse('2016T12:30:47.760-07:00') === Date.parse('2016T12:30:47.760738998-07:00'));
-+// CHECK-NEXT: true
+// CHECK-NEXT: true
+
+print(Date.parse('2016/01T12:30Z'));
+// CHECK-NEXT: 1451651400000
+print(Date.parse('2016/01/01T12:30Z'));
+// CHECK-NEXT: 1451651400000
+print((new Date('2017/02/15 15:01:37.243')).getTime());
+// CHECK-NEXT: 1487196097243
+print((new Date('2017/02/15 15:01:37.243')).getTime());
+// CHECK-NEXT: 1487196097243
+print((new Date('2017/02/15/15:01:37.243')).getTime());
+// CHECK-NEXT: 1487196097243
+print((new Date('2017/02/15T15:01:37.243')).getTime());
+// CHECK-NEXT: 1487196097243
+print((new Date('2017/02/15 15:01:37.243-0700')).getTime());
+// CHECK-NEXT: 1487196097243
+print((new Date('2017/02/15 15:01:37.243-07:00')).getTime());
+// CHECK-NEXT: 1487196097243
+print((new Date('2017/02/15 15:01:37.243-7:0')).getTime());
+// CHECK-NEXT: 1487196097243
+print((new Date('2017/02/15 15:01:37.243-0700')).getHours());
+// CHECK-NEXT: 15
+print((new Date('2017/02/15 15:01:37.243-0700')).getMinutes());
+// CHECK-NEXT: 1
+print((new Date('2017/02/15 15:01:37.243-0700')).getSeconds());
+// CHECK-NEXT: 37
+print((new Date('2017/02/15/15/01:37.243-07:00')).getTime());
+// CHECK-NEXT: NaN
+print((new Date('2017/02/15 15:01:37.243-07')).getTime());
+// CHECK-NEXT: 1487196097243
+print((new Date('2017/02/15 15:01:37.243-7')).getTime());
+// CHECK-NEXT: 1487196097243
+print((new Date('2017/02/15 15:01:37.243-7:')).getTime());
```

Comment:
I'm concerned about these trailing `:` formats. Is this really an expected working case? It seems malformed and that we should return `NaN` here.

Reply:
> I'm concerned we're being too permissive to the point that the semantics of `Date.parse` become too vague. Returning an unintended time can be more harmful and harder to debug than failing to parse and returning `NaN` for ambiguous input, particularly if we don't have extremely thorough documentation on non-obvious behavior. In the current state of the code on master, the permissible input is documented in the ECMAScript spec, with the only divergence being that certain minor delimiters are considered optional.
> 
> Browsers already support all kinds of different formats, but they are forced to hold a certain level of legacy compatibility which Hermes, as a newer engine, is not really obligated to.
> 
> I don't disagree with the spirit of this change, in that multiple of the new formats have readily apparent ways of understanding how they should be parsed, but I want to make sure that we don't confuse users and increase bugs because Hermes suddenly became too permissive in its Date parsing methods.
> 
> There are third-party libraries for parsing Dates in a more structured and well-documented format, so if a user wants to use those, the behavior is more easily understood.
> 
> Specific concerns are in the line comments, but in general, I think we should cut back a bit on the amount of possible inputs we're parsing in this PR.

I understand your concern.
Some of my changes were really too aggressive. I'll revert my changes and resubmit new PR that 
 isn't too permissive and support well-documented format. It is necessary to correct Date.parse on 
specified case. Because our business was done and invoked on JSC before. Switch to Hermes and we don't want to make big change to use third-party libraries. 
As for modifying some unittest's result,  I just want to keep logic of dateParsing consistant with JSC in the same situation.